### PR TITLE
[M68K] Add python binding support

### PR DESCRIPTION
**Pull request by nplanel**
Implement python binding using ctypes.
Add a basic test unit : test_m68k.py

Signed-off-by: Nicolas PLANEL <nplanel@gmail.com>

### CODE REVIEW

File: bindings/python/PKG-INFO [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Metadata-Version: 1.3
```
Comment (emoon) [inline bindings/python/PKG-INFO]: Is this file needed?

File: bindings/python/test_m68k.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    # "data" instruction generated by SKIPDATA option has no detail
+    if insn.id == 0:
+        return
+
+    if len(insn.operands) > 0:
+        print("\top_count: %u" % len(insn.operands))
+        c = -1
+        for i in insn.operands:
+            c += 1
+            if i.address_mode == M68K_OP_REG:
```
Comment (emoon) [inline bindings/python/test_m68k.py]: This code is incorrect. address_mode is of type https://github.com/emoon/capstone/blob/next/include/capstone/m68k.h#L49

I have being to want to fix the https://github.com/emoon/capstone/blob/next/include/capstone/m68k.h#L83 is actually being set on each op which isn't done today

File: bindings/python/PKG-INFO (hunk 1 above)
Comment (nplanel) [inline bindings/python/PKG-INFO]: no, my bad, I should change PKG-INFO.src and PKG-INFO.win
Could you reject the file from the PR, or do you want a new one ?

Comment (emoon) [inline bindings/python/PKG-INFO]: I think it's possible to rebase the PR without the change.

File: bindings/python/test_m68k.py (hunk 2 above)
Comment (nplanel) [inline bindings/python/test_m68k.py]: Yes I seen that, 30 min ago, I was cheated by the debugging output.
I will redo a new PR with the changes on this file tomorrow

Comment (emoon) [inline bindings/python/test_m68k.py]: :+1: 

Something that would be good to fix that all the build functions also set which type it is (a new var in m68k_op struct) that has this information. The problem is that in some cases it's not so simple.  For example movem has a bitfield of registers but perhaps https://github.com/emoon/capstone/blob/next/include/capstone/m68k.h#L77 should just be moved there instead and address_mode should always match the 68k one which would make sense.

Comment (nplanel) [inline bindings/python/test_m68k.py]: Yes think it's make more sense to add back the m68k_op_type field (simple) as the other arch have the same behaviour, meanwhile we can change the field name in cs_m68k_op to detail for m68k_adress_mode (mistyped)

Do that make sense for you ?

Comment (emoon) [inline bindings/python/test_m68k.py]: I suggest the struct should look something like this

```C
typedef struct cs_m68k_op {
	union {
		uint64_t imm;		// immediate value for IMM operand
		double fimm;
		m68k_reg reg;	// register value for REG operand
		m68k_op_mem mem;
	};
        m68k_op_type type;
	m68k_adress_mode address_mode;
} cs_m68k_op;
```

Then users of the struct knows if the op is mem, reg, imm, etc but address_mode is still here because this is needed to know if "a0" register is used as -(a0), (a0)+ etc

Comment (nplanel) [inline bindings/python/test_m68k.py]: Yes, to be consistent with other arch :+1: 

typedef struct cs_m68k_op {
    m68k_op_type type;
    union {
        uint64_t imm;       // immediate value for IMM operand
        double fimm;
        m68k_reg reg;   // register value for REG operand
        m68k_op_mem mem;
    };

    m68k_address_mode address_mode;
} cs_m68k_op;

Comment (emoon) [inline bindings/python/test_m68k.py]: Alright. I will fix this (hopefully tonight but not sure yet)

File: bindings/python/capstone/__init__.py [hunk 3]
```diff
@@ -106,13 +107,14 @@
...
-CS_ARCH_XCORE = 7
-CS_ARCH_MAX = 8
+CS_ARCH_M68K = 2
+CS_ARCH_MIPS = 3
+CS_ARCH_X86 = 4
+CS_ARCH_PPC = 5
+CS_ARCH_SPARC = 6
+CS_ARCH_SYSZ = 7
+CS_ARCH_XCORE = 8
+CS_ARCH_MAX = 9
```
Comment (nplanel) [inline bindings/python/capstone/__init__.py]: This would break the retro compatibility, the order should be preserved
The same order/id should be applied in cs_arch include/capstone/capstone.h

The result must be :
```python
CS_ARCH_ARM = 0
CS_ARCH_ARM64 = 1
CS_ARCH_MIPS = 2
CS_ARCH_X86 = 3
CS_ARCH_PPC = 4
CS_ARCH_SPARC = 5
CS_ARCH_SYSZ = 6
CS_ARCH_XCORE = 7
CS_ARCH_M68K = 8
CS_ARCH_MAX = 9
```
